### PR TITLE
Dump ConnectionManager state on SIGUSR1

### PR DESCRIPTION
**Pull request by coot**
- connection-manager: added readState method
- diffusion: dump connection manager state on SIGUSR1

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Diffusion.hs [hunk 1]
```diff
@@ -816,7 +824,19 @@ runDataDiffusion tracers
                   connectionHandler
                   classifyHandleError
                   NotInResponderMode
-                  $ \(connectionManager :: NodeToNodeConnectionManager InitiatorMode Void) ->
+                  $ \(connectionManager :: NodeToNodeConnectionManager InitiatorMode Void) -> do
+#ifdef POSIX
+                  _ <- Signals.installHandler
```
Comment (karknu) [inline ouroboros-network/src/Ouroboros/Network/Diffusion.hs:829]: Remove the space after `installHandler`.

File: ouroboros-network/src/Ouroboros/Network/Diffusion.hs [hunk 2]
```diff
@@ -923,7 +943,18 @@ runDataDiffusion tracers
                   connectionHandler
                   classifyHandleError
                   (InResponderMode controlChannel)
-                  $ \(connectionManager :: NodeToNodeConnectionManager InitiatorResponderMode ()) ->
+                  $ \(connectionManager :: NodeToNodeConnectionManager InitiatorResponderMode ()) -> do
+#ifdef POSIX
+                  _ <- Signals.installHandler
```
Comment (karknu) [inline ouroboros-network/src/Ouroboros/Network/Diffusion.hs:948]: and here.

**Review by bolt12 — APPROVED** [review]
LGTM!

File: ouroboros-network/src/Ouroboros/Network/Diffusion.hs (hunk 1 above)
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Diffusion.hs]: Done

File: ouroboros-network/src/Ouroboros/Network/Diffusion.hs (hunk 2 above)
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Diffusion.hs]: Done

**Review by karknu — APPROVED** [review]
LGTM

**Pull request merged**